File: 36-valid-sudoku/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {


        Set<String> set = new HashSet();
        


        for(int i=0;i<9;i++){

            for(int j=0;j<9;j++){


                char ch = board[i][j];

                if(ch!='.'){


                    if(!(set.add(ch+" is present in row"+i)) || !(set.add(ch+" is present in column"+j)) || !(set.add(ch+" is present in block"+i/3+"-"+j/3))){
                        return false;
                    }


                }




            }


        }
        return true;
        
        

        
    }
}

File: 36-valid-sudoku/valid-sudoku.ts
function isValidSudoku(board: string[][]): boolean {
    

    const set = new Set()

    for(let i=0;i<9;i++){
        for(let j=0;j<9;j++){

            const cell = board[i][j]

            if(cell==='.') continue

            const row=`${cell} is present in row ${i}`
            const column=`${cell} is present in column ${j}`
            const box=`${cell} is present in block ${Math.floor(i/3)} - ${Math.floor(j/3)}`

            if(set.has(row) || set.has(column) || set.has(box)) return false

            set.add(row).add(column).add(box)

        


        }
    }

    return true
};